packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -124,33 +125,43 @@ class Checkbox extends StatefulWidget {
   ///
   /// When [value] is null and [tristate] is true, [MaterialState.selected] is
   /// included as a state.
+  /// {@endtemplate}
   ///
-  /// If this property is null, [MaterialStateMouseCursor.clickable] will be used.
+  /// If null, then the value of [CheckboxThemeData.mouseCursor] is used. If
+  /// that is also null, then [MaterialStateMouseCursor.clickable] is used.
   final MouseCursor? mouseCursor;
 
   /// The color to use when this checkbox is checked.
   ///
   /// Defaults to [ThemeData.toggleableActiveColor].
   ///
-  /// If [fillColor] returns a non-null color in the [MaterialState.selected]
-  /// state, it will be used instead of this color.
+  /// If [fillColor] or [CheckboxThemeData.fillColor] returns a non-null color
+  /// in the [MaterialState.selected] state, it will be used instead of this
+  /// color.
   final Color? activeColor;
 
+  /// {@template flutter.material.checkbox.fillColor}
   /// The color that fills the checkbox when it is checked, in all
   /// [MaterialState]s.
   ///
-  /// If this is provided, it will be used over [activeColor].
-  ///
   /// Resolves in the following states:
   ///  * [MaterialState.selected].
   ///  * [MaterialState.hovered].
   ///  * [MaterialState.focused].
   ///  * [MaterialState.disabled].
+  /// {@endtemplate}
+  ///
+  /// If null, then the value of [CheckboxThemeData.fillColor] is used. If that
+  /// is also null, then the value of [activeColor] is used in the selected
+  /// state.
   final MaterialStateProperty<Color?>? fillColor;
 
+  /// {@template flutter.material.checkbox.checkColor}
   /// The color to use for the check icon when this checkbox is checked.
+  /// {@endtemplate}
   ///
-  /// Defaults to Color(0xFFFFFFFF)
+  /// If null, then the value of [CheckboxThemeData.checkColor] is used. If
+  /// that is also null, then Color(0xFFFFFFFF) is used.
```

Comment:
Instead of using fff use the theme.colorScheme.background or something like that. So it works on dark and light mode.

Reply:
I don't want to make any (potentially breaking) visual changes in this PR. I think a separate PR can handle this.